include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -298,114 +364,93 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _ReduceOp __reduce_op, _T
     auto __n = oneapi::dpl::__ranges::__get_first_range_size(__rngs...);
     assert(__n > 0);
 
-    // Use a single-task sequential implementation for very small arrays.
-    if (__n <= 64)
-    {
-        return __parallel_transform_reduce_seq_impl<_Tp>(::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op,
-                                                         __transform_op, __init, ::std::forward<_Ranges>(__rngs)...);
-    }
-
-    // Use a single-pass tree reduction for medium-sized arrays with the following template parameters:
-    // __iters_per_work_item shows number of elements to reduce on global memory.
-    // __work_group_size shows number of elements to reduce on local memory.
-
-    // get the work group size adjusted to the local memory limit
-    // Pessimistically double the memory requirement to take into account memory used by compiled kernel
-    // TODO: find a way to generalize getting of reliable work-group size
+    // Get the work group size adjusted to the local memory limit.
+    // Pessimistically double the memory requirement to take into account memory used by compiled kernel.
+    // TODO: find a way to generalize getting of reliable work-group size.
     ::std::size_t __work_group_size = oneapi::dpl::__internal::__slm_adjusted_work_group_size(__exec, sizeof(_Tp) * 2);
 
-    if (__n <= 65536 && __work_group_size >= 512)
+    // Use single work group implementation if array < __work_group_size * __iters_per_work_item.
+    if (__n <= 256 && __work_group_size >= 256)
```

Comment:
Would it be a better practice to turn all subsequent `if` statements into `else if` statements?

Reply:
I don't have a strong opinion on this but `else if` might be easier to read.

Reply:
Doesn't `else if` avoid a lot of unnecessary branching for a single control flow (once you hit your `true` case, you skip all the rest of the branching)?
Unless we have a explicit need to have consecutive `if` statements, I think `else if` is correct.

Reply:
While I agree in general, this routine uses return statements in the if statements.

Reply:
Ah yes, my mistake.  I'll still put my vote in for `else if` because that is the norm here I think.
However, I'll retract my previous statement.

Reply:
I switched to `else if` statements.

Reply:
Also for this set of `if`/`else if` statements:  They are all checking `__work_group_size >=256`, does it make sense to pull this out and above?